src/player.js

```diff
@@ -635,7 +636,36 @@ export default class Player extends FakeEventTarget {
     this.dispatchEvent(new FakeEvent(CustomEventType.PLAYER_DESTROY));
     this._eventManager.destroy();
   }
+  /**
+   * Listen to the video element events and triggers them from the engine.
+   * @public
+   * @returns {void}
+   */
+  attachMediaSource(): void {
+    if (this._engine) {
+      this._engine.attachMediaSource();
+    }
+  }
 
+  /**
+   * Listen to the video element events and triggers them from the engine.
+   * @public
+   * @returns {void}
+   */
+  detachMediaSource(): void {
+    if (this._engine) {
+      this._engine.detachMediaSource();
+    }
+  }
+  handleAdsWithMSE(): void {
+    if (this.config.playback.playAdsWithMSE) {
+      this._eventManager.listen(this, AdEventType.AD_BREAK_START, this.detachMediaSource);
+      this._eventManager.listen(this, AdEventType.AD_BREAK_END, this.attachMediaSource);
+      this._eventManager.listen(this, AdEventType.AD_ERROR, this.attachMediaSource);
```

Comment:
need to better understand this, in general `AD_BREAK_END` should be fired if ad was loaded and then failed.
Maybe you are referring to a case where the initial ad request failed and video ad never loaded(i.e. also `AD_BREAK_START` never happened) and in this case is this tested and you know this is needed?